src/org/opendatakit/briefcase/ui/reused/Analytics.java

```diff
@@ -0,0 +1,127 @@
+    this.userId = userId;
+    this.screenSize = screenSize;
+    this.windowSizeGetter = windowSizeGetter;
+  }
+
+  public static Analytics from(String trackingId, String version, String userId, Dimension screenSize, Supplier<Dimension> windowSizeGetter) {
+    GoogleAnalytics ga = GoogleAnalytics.builder()
+        .withTrackingId(trackingId)
+        .build();
+    return new Analytics(ga, version, userId, buildResolutionString(screenSize), windowSizeGetter);
+  }
+
+  public void enter(String screenName) {
+    buildScreenViewHit().screenName(screenName).sessionControl("start").sendAsync();
+  }
+
+  public void leave(String screenName) {
+    buildScreenViewHit().screenName(screenName).sessionControl("end").sendAsync();
+  }
+
+  public void event(String category, String action, String label, Integer value) {
+    EventHit request = buildEventHit()
+        .eventCategory(category)
+        .eventAction(action);
+    Optional.ofNullable(label).ifPresent(request::eventLabel);
+    Optional.ofNullable(value).ifPresent(request::eventValue);
+    request.sendAsync();
+  }
+
+  public ComponentListener buildComponentListener(String screenName) {
+    return new ComponentListener() {
+      @Override
+      public void componentResized(ComponentEvent e) {
+      }
+
+      @Override
+      public void componentMoved(ComponentEvent e) {
+      }
+
+      @Override
+      public void componentShown(ComponentEvent e) {
+        enter(screenName);
+      }
+
+      @Override
+      public void componentHidden(ComponentEvent e) {
+        leave(screenName);
+      }
+    };
+  }
+
+  public void enableTracking() {
+    event("Settings", "Track consent", "Enable", null);
+    ga.getConfig().setEnabled(true);
+    System.out.println("ENABLE TRACKING");
+  }
+
+  public void disableTracking() {
+    event("Settings", "Track consent", "Disable", null);
+    ga.getConfig().setEnabled(false);
+    System.out.println("DISABLE TRACKING");
+  }
+
+  public void register(JComponent component) {
+    component.addComponentListener(buildComponentListener("Export"));
+
+  }
+
+  private EventHit buildEventHit() {
+    return ga.event()
+        .applicationVersion(version)
+        .applicationName("ODK Briefcase")
+        .userId(userId)
+        .screenResolution(screenSize)
+        .anonymizeIp(true)
+        .userAgent(USER_AGENT)
+        .dataSource("app")
+        .viewportSize(buildResolutionString(windowSizeGetter.get()));
+  }
+
+  private ScreenViewHit buildScreenViewHit() {
+    return ga.screenView()
+        .applicationVersion(version)
+        .applicationName("ODK Briefcase")
+        .userId(userId)
+        .screenResolution(screenSize)
+        .anonymizeIp(true)
+        .userAgent(USER_AGENT)
+        .dataSource("app")
+        .viewportSize(buildResolutionString(windowSizeGetter.get()));
+  }
+
+  private static String buildResolutionString(Dimension dimension) {
+    return String.format("%sx%s", dimension.width, dimension.height);
```

Comment:
I wonder if whatever takes this n × n string would like a real times/by character (**×**) instead of a crude, primitive x. I like to think that some people somewhere appreciate these fine touches.

Reply:
I think GA requires an `x` :(

You're right with the %d... I use %s as a reflex...